src/pages/ReimbursementAccount/ReimbursementAccountPage.js

```diff
@@ -97,438 +97,389 @@ const defaultProps = {
+            // Since there is no VBBA in progress, we won't need to show the component ContinueBankAccountSetup
+            return false;
         }
+        return achData.state === BankAccount.STATE.PENDING || _.contains([CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT, ''], getStepToOpenFromRouteParams());
     }
 
-    /**
-     * Returns true if a VBBA exists in any state other than OPEN or LOCKED
-     * @returns {Boolean}
-     */
-    hasInProgressVBBA() {
-        const achData = lodashGet(this.props.reimbursementAccount, 'achData', {});
-        return achData.bankAccountID && achData.state !== BankAccount.STATE.OPEN && achData.state !== BankAccount.STATE.LOCKED;
-    }
+    const [hasACHDataBeenLoaded, setHasACHDataBeenLoaded] = useState(reimbursementAccount !== ReimbursementAccountProps.reimbursementAccountDefaultProps);
+    const [shouldShowContinueSetupButton, setShouldShowContinueSetupButton] = useState(
+        hasACHDataBeenLoaded ? getShouldShowContinueSetupButtonInitialValue() : false
+    );
+    
+    const currentStep = achData.currentStep || CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
+    const policyName = lodashGet(policy, 'name', '');
+    const policyID = lodashGet(route.params, 'policyID', '');
+    const { translate } = useLocalize();
+    const { isOffline } = useNetwork();
+    const prevIsOfflineRef = useRef(isOffline);
+    const prevReimbursementAccountRef = useRef(reimbursementAccount);
+    const requestorStepRef = useRef(null);
 
     /**
      * Retrieve verified business bank account currently being set up.
      * @param {boolean} ignoreLocalCurrentStep Pass true if you want the last "updated" view (from db), not the last "viewed" view (from onyx).
      */
-    fetchData(ignoreLocalCurrentStep) {
+    function fetchData(ignoreLocalCurrentStep) {
         // Show loader right away, as optimisticData might be set only later in case multiple calls are in the queue
         BankAccounts.setReimbursementAccountLoading(true);
 
         // We can specify a step to navigate to by using route params when the component mounts.
         // We want to use the same stepToOpen variable when the network state changes because we can be redirected to a different step when the account refreshes.
-        const stepToOpen = this.getStepToOpenFromRouteParams();
-        const achData = lodashGet(this.props.reimbursementAccount, 'achData', {});
+        const stepToOpen = getStepToOpenFromRouteParams();
         const subStep = achData.subStep || '';
         const localCurrentStep = achData.currentStep || '';
         BankAccounts.openReimbursementAccountPage(stepToOpen, subStep, ignoreLocalCurrentStep ? '' : localCurrentStep);
-    }
+    };
+
+    // Will run whenever reimbursementAccount prop changes, and update the hasACHDataBeenLoaded state if necessary
+    useEffect(() => {
+        setHasACHDataBeenLoaded(reimbursementAccount !== ReimbursementAccountProps.reimbursementAccountDefaultProps);
+    }, [reimbursementAccount]);
```

Comment:
I believe its origin in the original class component's code is
`const hasACHDataBeenLoaded = 
    this.props.reimbursementAccount !== ReimbursementAccountProps.reimbursementAccountDefaultProps;`
    
In a class component, hasACHDataBeenLoaded's value was re-evaluated with every render based on the current value of the reimbursementAccount prop.

In the function component, the hasACHDataBeenLoaded value is a piece of state. Instead of being re-evaluated with every render directly, it's updated through the setHasACHDataBeenLoaded function within a useEffect. This useEffect runs whenever there are changes to the reimbursementAccount prop.

Reply:
NAB, because I know we are not overhauling the logic here. But it's unclear to me why this would need to be in state. Is it possible to work with `useMemo()` instead to set an initial value for `hasACHDataBeenLoaded` and later update it when it loads?